src/app/shared/services/motor.service.ts

```diff
@@ -14,6 +14,9 @@ import {MotorSettingsError} from "../error/motor-settings-error";
     providedIn: "root",
 })
 export class MotorService {
+    selectedMotorName: BehaviorSubject<string> = new BehaviorSubject<string>(
+        "",
+    );
```

Comment:
since this subject is only used for parent-child-communication between motor-position-component and joint-control-core component, it should be replaced with an event, that the motor-position-component emits and the joint-control-core component listens to. Alternatively, we could replace the `[routerLinks]`s in the joint-control-core component with `(click)`-events. In the callback-function of the click-events we could directly select the motor and then use `router.navigate(...)` to route to the selected motor. This way, we could omit the roundtrip to the motor-position-component